syncapi/sync/requestpool.go

```diff
@@ -74,6 +74,14 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *userapi.
 		"limit":     syncReq.limit,
 	})
 
+	lsreq := &userapi.PerformLastSeenUpdateRequest{
+		UserID:     device.UserID,
+		DeviceID:   device.ID,
+		RemoteAddr: req.RemoteAddr,
+	}
+	lsres := &userapi.PerformLastSeenUpdateResponse{}
+	go rp.userAPI.PerformLastSeenUpdate(req.Context(), lsreq, lsres) // nolint:errcheck
```

Comment:
We probably want to add some sort of rate-limiting here, we don't need millisecond accuracy on the last seen timestamp, heck even just by the minute would do. I would be tempted to:
 - Have a sync map of deltas (e.g `[user_id + device_id] => []remoteAddrs`)
 - Have a background worker which periodically (every 1 minute?) purges the map.

The danger of this current solution is that I have no confidence that we don't tight loop under some cases, which will cause a _lot_ of goroutines to be made, resources consumed, etc.